common/include/common/refcount.h

```diff
@@ -0,0 +1,30 @@
+/*
+Looking Glass - KVM FrameRelay (KVMFR) Client
+Copyright (C) 2021 Guanzhong Chen <quantum2048@gmail.com>
+https://looking-glass.hostfission.com
+
+This program is free software; you can redistribute it and/or modify it under
+the terms of the GNU General Public License as published by the Free Software
+Foundation; either version 2 of the License, or (at your option) any later
+version.
+
+This program is distributed in the hope that it will be useful, but WITHOUT ANY
+WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A
+PARTICULAR PURPOSE. See the GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License along with
+this program; if not, write to the Free Software Foundation, Inc., 59 Temple
+Place, Suite 330, Boston, MA 02111-1307 USA
+*/
+
+#include <stddef.h>
+
+struct CountedBuffer {
+  _Atomic(size_t) refs;
+  size_t size;
+  char * data;
+};
+
+struct CountedBuffer * countedBufferCreate(size_t size);
+void countedBufferIncRef(struct CountedBuffer * buffer);
+void countedBufferDecRef(struct CountedBuffer * buffer);
```

Comment:
I am not sure I like the naming used here, I would suggest it follows that which is used by other referenced counting APIs like OLE/COM. Ie, `AddRef` and `Release`.

Reply:
Changed to COM-style `AddRef` and `Release`.